tests/out/hlsl/array-in-ctor.yaml

```diff
@@ -0,0 +1,5 @@
+vertex: []
+fragment: []
+compute:
```

Comment:
suggestion: Stylistically, it's entirely valid to omit `vertex` and `fragment`. Shall we strip out empty shader model categories?

Reply:
Done, as of 7fac4866.

Reply:
Hmm, using `serde_with` to achieve this adds more than 20 build steps, and significantly adds to compile times and compile lengths by bringing in its own proc-macros and the `darling` ecosystem. Not sure if this is worth it. 🤔